src/schema/v2/artwork/index.ts

```diff
@@ -208,6 +245,24 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
       },
       marketPriceInsights: {
         type: ArtworkPriceInsightsType,
+        resolve: async (
+          { artist, medium, category },
+          _,
+          { marketPriceInsightsBatchLoader }
+        ) => {
+          if (!marketPriceInsightsBatchLoader) return false
+
+          const marketPriceInsightNodes = await marketPriceInsightsBatchLoader([
+            {
+              artistId: artist._id,
+              medium,
+              category,
+            },
+          ])
+
+          if (marketPriceInsightNodes.length === 0) return null
+          return marketPriceInsightNodes[0]
+        },
```

Comment:
**(issue):** Wouldn't this cause a request per artwork when loading multiple artworks? 

The reason why we introduced the batch load in the first place was to be able to [load all price insights at once](src/schema/v2/me/myCollection.ts).

We could try to do the same in the [artwork resolver](src/schema/v2/artwork/index.ts) and load price insights whenever `marketPriceInsights` is requested for artworks.

Reply:
true, i am actually working on fixing that right now and how I can avoid it. thanks for also pointing it

Reply:
The thing is that I would like to also be able to request the price insights for an artwork without using the global marketPriceInsights interface and therefore the need to get the medium first and make the query and ending up nesting query renderers

Reply:
I've fixed the links in the comment above ⬆️ .

You could load the artwork first and then enrich with the price insights (if the field is requested) [here](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/artwork/index.ts#L1460)